### ISSUE TITLE
mobx.useStrict causes errors

### ISSUE THREAD

**Issue by dargue3** [mobxjs/mobx-state-tree#494]
I noticed when you include the line `mobx.useStrict(true)`, you get errors about needing to wrap changes to observables in an action. I included the strict mode just assuming that it would warn devs on my team if they used anti-patterns. My code looks like this:

```javascript
const Product = t
  .model({
    id: t.identifier(t.number),
    name: t.string,
    brandName: t.string,
  });

const Products = t
  .model('Products', {
    isLoading: true,
    productsMap: t.map(Product),
  })
  .actions((self) => {
    function addProduct(p) {
      self.productsMap.put(p);
    }

    function markLoading(state) {
      self.isLoading = state;
    }

    const fetchById = flow(function* (id) {
      markLoading(true);
      const req = yield api('GET_PRODUCTS_BY_ID', { id });
      addProduct(req[0]);
      markLoading(false);
    });

    return { fetchById };
  });
```

When running `store.products.fetchById(id)`, I would encounter an error like:

```[mobx] Invariant failed: Since strict-mode is enabled, changing observed observable values outside actions is not allowed.```

Is the advice here just to leave the errors about strictness up to MST? Or is this a bug?

(running mobx `3.3.1` and mobx-state-tree `1.1.0`)

**Comment by mweststrate** [comment]
Should work the same. Maybe generators are transpiled differently in your
server setup?

**Comment by mikecann** [comment]
I thought I read somewhere that useStrict cant be used with MST?

**Comment by codyhatch** [comment]
I'm also experiencing this.  Would love to know any workarounds or solutions.

**Comment by mikecann** [comment]
@codyhatch I just commented it out for now 😢

**Comment by mweststrate** [comment]
Can anybody provide small reproduction?

**Comment by mweststrate** [comment]
Pretty sure this is fixed in 1.3.1. Let us know if that isn't the case!
